pkg/service/workflowsvc/service.go

```diff
@@ -2,22 +2,91 @@ package workflowsvc
 
 import (
 	"context"
+	"time"
 
 	"github.com/benbjohnson/clock"
 	"github.com/common-fate/apikit/logger"
 	"github.com/common-fate/common-fate/pkg/access"
+	"github.com/common-fate/common-fate/pkg/cache"
 	"github.com/common-fate/common-fate/pkg/gevent"
 	"github.com/common-fate/common-fate/pkg/storage"
 	"github.com/common-fate/common-fate/pkg/types"
 	"github.com/common-fate/ddb"
+	"github.com/common-fate/iso8601"
+	"go.uber.org/zap"
 )
 
+// copy of grouptarget to be used in the granting process.
+// Uses iso8601 time for grant start and end to preserve timezone when granting
+type CreateGroupTargetRequest struct {
+	ID        string `json:"id" dynamodbav:"id"`
+	GroupID   string `json:"groupId" dynamodbav:"groupId"`
+	RequestID string `json:"requestId" dynamodbav:"requestId"`
+	// Also denormalised across all the request items
+	RequestStatus types.RequestStatus `json:"requestStatus" dynamodbav:"requestStatus"`
+	RequestedBy   access.RequestedBy  `json:"requestedBy" dynamodbav:"requestedBy"`
+	// The id of the cache.Target which was used to select this on the request.
+	// the cache item is subject to be deleted so this cacheID may not always exist in the future after the grant is created
+	TargetCacheID string         `json:"cacheId" dynamodbav:"cacheId"`
+	TargetGroupID string         `json:"targetGroupId" dynamodbav:"targetGroupId"`
+	TargetKind    cache.Kind     `json:"targetGroupFrom" dynamodbav:"targetGroupFrom"`
+	Fields        []access.Field `json:"fields" dynamodbav:"fields"`
+	// The grant will be populated when this target is submitted to be provisioned
+	// The start and end time are calculated and stored on the grant when it is provisioned
+	Grant     *WorkflowGrant `json:"grant" dynamodbav:"grant"`
+	CreatedAt time.Time      `json:"createdAt" dynamodbav:"createdAt"`
+	UpdatedAt time.Time      `json:"updatedAt" dynamodbav:"updatedAt"`
+	// request reviewers are users who have one or more groups to review on the request as a whole
+	RequestReviewers []string `json:"requestReviewers" dynamodbav:"requestReviewers, set"`
+}
+
+type WorkflowGrant struct {
+	// The user email
+	Subject string                               `json:"subject" dynamodbav:"subject"`
+	Status  types.RequestAccessGroupTargetStatus `json:"status" dynamodbav:"status"`
+	//the time which the grant starts
+	Start iso8601.Time `json:"start" dynamodbav:"start"`
```

Comment:
@meyerjrr is the only difference here that the grant has iso times?
I think it was probably just an oversight from me not using the iso package on the access.Grant struct.
Would you be able to just update that struct instead and not have the duplicate here?

Reply:
Tried that at first, but would save as nil to the dynamo table when saving the grant. Think this is why we followed this pattern in the old request flow?

Reply:
oh, this is possibly because the type doesn't implement a ddb marshalling method